storyblok/storyblok-js-client
# fix: get proper binding for the fetch function

Hey team, this issue is important as it makes js-client unusable. The fix is on the binding of `fetch` which is bound to the instance of the client instead of `window`. Check this stackoverflow thread for more info: https://stackoverflow.com/questions/44720448/fetch-typeerror-failed-to-execute-fetch-on-window-illegal-invocation

**Btw, I know the type `any` is not ideal** but as this is blocking we can improve that later.

## Pull request type


## How to test this PR

- Switch to the branch of this PR
- Open one terminal (terminal A) - run `npm run build` to ensure you're using latest js-client dist
- Open another terminal (terminal B) - run `cd playground-svelte`, run `npm i`, and run `npm run demo`. Open `http://localhost:5174/` in the browser and all should be good (as in image 1 below) - ISSUE FIXED
- Now in `src/sbFetch` line 33, change is as it was before `this.fetch = $c.fetch(args) || fetch(args)`, and in terminal A run `npm run build` again
- Reload the page on the browser, open the console, and you'll see the `fetch` function is failing (as in image 2 below) - CURRENT ISSUE

IMAGE 1 - ALL GOOD
<img width="1440" alt="Screenshot 2023-12-15 at 17 04 04" src="https://github.com/storyblok/storyblok-js-client/assets/5701162/a8e70dd8-c968-4c08-baf3-1ea232618f93">

IMAGE 2 - FETCH FAILING
<img width="1440" alt="Screenshot 2023-12-15 at 17 05 02" src="https://github.com/storyblok/storyblok-js-client/assets/5701162/54661233-cea3-4117-901d-d59ff940e093">
